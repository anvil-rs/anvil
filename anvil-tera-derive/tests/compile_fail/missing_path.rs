// This file tests the error when a template is missing the required path attribute
// Expected error: "Missing 'path' attribute within #[template(...)]"

use anvil_tera_derive::Template;
use serde::Serialize;
use std::sync::LazyLock;
use tera::Tera;

static TEMPLATES: LazyLock<Tera> = LazyLock::new(|| Tera::default());

#[derive(Serialize, Template)]
#[template(tera = TEMPLATES)]  // Missing path attribute
struct MissingPathTemplate {
    value: String,
}

fn main() {
    // This will never run, we're only testing the compile-time error
}
