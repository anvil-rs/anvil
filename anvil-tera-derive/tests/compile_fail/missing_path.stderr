error: Missing 'path' attribute within #[template(...)]. Example: #[template(path = "my_template.html", ...)]
  --> $DIR/missing_path.rs:11:1
   |
11 | #[derive(Serialize, Template)]
   | ^
